File: C++/007.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N, Q;
    cin >> N;
    vector<int> A(N);
    for (int i = 0; i < N; i++) cin >> A[i];
    cin >> Q;
    vector<int> B(Q);
    for (int i = 0; i<Q; i++) cin >> B[i];

    sort(A.begin(),A.end());

    for (int i = 0; i<Q; i++) {
        if (B[i] > A[N-1]) {
            cout << B[i] - A[N-1] << endl;
        }
        else {
            auto position = lower_bound(A.begin(),A.end(), B[i]);
            if (position == A.begin()) {
                cout << *position - B[i] << endl;
            }
            else {
                cout << min(abs(*position - B[i]), abs(*(position-1)-B[i])) << endl;
            }
        }
    }
}

File: Python/007.py
def binary_search(data, value):
    left = 0 
    right = len(data) - 1 
    while left <= right:
        mid = (left + right) // 2
        if value == data[mid]:
            return mid
        elif value > data[mid]:
            left = mid + 1  
        else:
            right = mid - 1
    return left


N = int(input())
A = list(map(int, input().split()))

A.sort()

Q = int(input())
B = []
for i in range(Q):
    B.append(int(input()))

for i in range(Q):
    left = binary_search(A, B[i])
    if left == len(A):
        print(B[i] - A[left-1])
    elif left == 0:
        print(abs(A[left] - B[i]))
    else:
        print(min(abs(A[left] - B[i]), abs(B[i] - A[left-1])))